packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ipv6/ipv6.service.js

```diff
@@ -0,0 +1,28 @@
+import { PRODUCT_TYPES } from './ipv6.constant';
+
+export default class IpAgoraV6Order {
+  /* @ngInject */
+  constructor($http, IpAgoraOrder) {
+    this.$http = $http;
+    this.IpAgoraOrder = IpAgoraOrder;
+  }
+
+  getVrackService() {
+    return this.$http
+      .get('/products', {
+        params: {
+          product: PRODUCT_TYPES.vrack.apiTypeName,
+        },
+        serviceType: 'aapi',
+      })
+      .then(({ data }) => {
+        const availableVrack = this.IpAgoraOrder.handleErrorOrServices(data);
+        return [
+          ...availableVrack.map((vrackService) => ({
+            ...vrackService,
+            type: PRODUCT_TYPES.vrack.apiTypeName,
+          })),
+        ];
+      });
```

Comment:
```suggestion
    return this.iceberg('/vrack')
    .query()
    .expand('CachedObjectList-Pages')
    .execute()
    .$promise
    .then(({ data }) => {
      return [
        ...data.map((vrackService) => {
          const serviceName = vrackService.iam.urn.split(':').pop();
          const displayName = vrackService.name || serviceName;
          return {
            displayName,
            serviceName,
            type: PRODUCT_TYPES.vrack.apiTypeName,
          }
        }),
      ];
    });
```
You can call APIv6 directly with Iceberg to retrieve the vrack of the user